spec/controllers/ems_cloud_controller_spec.rb

```diff
@@ -0,0 +1,33 @@
+require "spec_helper"
+
+describe EmsCloudController do
+  describe "#create" do
+    before do
+      EvmSpecHelper.seed_specific_product_features("ems_cloud_new")
+      feature = MiqProductFeature.find_all_by_identifier(["ems_cloud_new"])
+      test_user_role  = FactoryGirl.create(:miq_user_role,
+                                           :name                 => "test_user_role",
+                                           :miq_product_features => feature)
+      test_user_group = FactoryGirl.create(:miq_group, :miq_user_role => test_user_role)
+      user = FactoryGirl.create(:user, :name => 'test_user', :miq_groups => [test_user_group])
+
+      allow(user).to receive(:server_timezone).and_return("UTC")
+      described_class.any_instance.stub(:set_user_time_zone)
+      controller.stub(:check_privileges).and_return(true)
+      login_as user
+    end
+
+    it "adds a new provider" do
+      controller.instance_variable_set(:@breadcrumbs, [])
+      lambda do
+        controller.new
+      end.should_not raise_error
+      controller.instance_variable_set(:@edit, :new => {:name => "EMS 1", :emstype => "rhevm", :hostname => nil},
```

Comment:
@lgalis you don't need to setup session[:edit] & @edit here and in other 2 files, these get set by "new" method call, instead you can add a check after get :new something like this:
expect(assigns(:ems)).to_not be_nil
expect(assigns(:edit)).to_not be_nil